### PR TITLE
Just for stealth-testing my fork in your CI

### PR DESCRIPTION
**Pull request by DaniloE**
You can ignore it if you want ;)

### CODE REVIEW

File: storage/backends/local/local.go [hunk 1]
```diff
@@ -21,37 +21,93 @@
...
 	log.Debug("Get backups from folder: ", viper()["backupdir"])
-	backups.GetBackupsInDir(viper()["backupdir"].(string))
-	backups.WalDir = filepath.Join(viper()["archivedir"].(string), subDirWal)
-	return backups
+	backupDir := viper()["backupdir"].(string)
+	files, _ := ioutil.ReadDir(backupDir)
+	for _, f := range files {
+		var newBackup backup.Backup
+		var err error
+		path := backupDir + "/" + f.Name()
```
Comment (xxorde) [inline storage/backends/local/local.go]: Maybe use filepath.Join() instead

File: storage/backends/local/local.go [hunk 2]
```diff
@@ -172,3 +228,89 @@ func (b Localbackend) GetBasebackup(viper func() map[string]interface{}, backup
...
+	// find all backup labels
+	for _, f := range files {
+		if f.Size() > backup.MaxBackupLabelSize {
+			// size is to big for backup label
+			continue
+		}
+		if backup.RegBackupLabelFile.MatchString(f.Name()) {
+			log.Debug(f.Name(), " => seems to be a backup Label, by size and name")
+
+			labelFile := bp.Backups.WalPath + "/" + f.Name()
```
Comment (xxorde) [inline storage/backends/local/local.go]: Maybe use filepath.Join() instead

File: storage/backends/s3/s3.go [hunk 3]
```diff
@@ -283,3 +347,133 @@ func (b S3backend) GetBasebackup(viper func() map[string]interface{}, backup *ut
...
+			count++
+		}
+
+	}
+	return count, err
+}
+
+// GetStartWalLocation returns the oldest needed WAL file
+// Every older WAL file is not required to use this backup
+func (b S3backend) GetStartWalLocation(bp *backup.Backup) (startWalLocation string, err error) {
```
Comment (xxorde) [inline storage/backends/s3/s3.go]: Do we need to implement the logic for every storage engine, or can we put this functionality somewhere else?

File: storage/backends/local/local.go (hunk 1 above)
Comment (DaniloE) [inline storage/backends/local/local.go]: ok

File: storage/backends/local/local.go (hunk 2 above)
Comment (DaniloE) [inline storage/backends/local/local.go]: ok

File: storage/backends/s3/s3.go (hunk 3 above)
Comment (DaniloE) [inline storage/backends/s3/s3.go]: I am working on this...this is one of the things i would like to change or make easier. Also i think the use of func() map... is not very sexy...

**Pull request merged**